Rakefile

```diff
@@ -10,8 +10,51 @@ namespace :design_system do
   end
 end
 
-desc "Rubocop Linting"
-task :rubocop do
-  puts "Running rubocop linter"
-  system("bundle exec rubocop") || raise
+desc "Run all checks in the local context"
+task check: ["ruby:lint", "npm:test"]
+
+namespace :ruby do
+  desc "Lint ruby files"
+  task lint: [:rubocop, :haml_lint]
+
+  desc "Rubocop Linting"
+  task :rubocop do
+    puts "Running rubocop linter"
+    system("bundle exec rubocop")
+  end
+
+  desc "Haml Linting"
+  task :haml_lint do
+    puts "Running haml-lint"
+    system("bundle exec haml-lint haml styleguide")
+  end
+end
+
+namespace :npm do
+  desc "Run node test"
+  task test: [:lint, :jest]
+
+  desc "Run jest tests in node"
+  task jest: :build do
+    puts "Running jest tests"
+    system("npm run jest")
+  end
+
+  desc "Run linting jobs in node"
+  task lint: :build do
+    puts "Running Node lint"
+    system("npm run lint")
+  end
+
+  desc "Install Node modules"
+  task :install do
```

Comment:
Is there a reason for dual dependencies, i.e. could these be ran separate. If not why don't we combine them as `npm install && npm build`

Reply:
Possibly - I can easily imagine other tasks which depend on install but not on build.